src/implementation.cpp

```diff
@@ -3,6 +3,11 @@
 #include <climits>
 #include <type_traits>
 
+static_assert(sizeof(uint8_t) == sizeof(char), "simdutf requires that uint8_t be a char");
```

Comment:
isn't this implicitly covered by the test on row 9 for CHAR_BIT ?

Reply:
Yes. It is for debugging purposes. I am adding a comment.

I am concerned with broken compiler systems. I want to catch them.